### PR TITLE
Cte tutorial

### LINKED ISSUE #50: Write tutorial: CTES
Cover:

Simple
Multiple
Update
Delete

operations with CTE

### PR DESCRIPTION
**Pull request by lfunderburk**
## Issue number

Closes #50 

## Checklist before requesting a review

- [x] Performed a self-review of my code
- [x] Formatted my code with [`pkgmt format`](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#linting-formatting)

----
:books: Documentation preview :books:: https://ploomber-sql--51.org.readthedocs.build/en/51/

### CODE REVIEW

**Comment by lfunderburk** [comment]
Hi @jpjon and @TegveerG 
CTE tutorial is up

File: course-material/advanced-querying-techniques/ctes.md [hunk 1]
```diff
@@ -6,9 +6,417 @@ jupytext:
...
+* The `UNION ALL` clause is used to combine the results of the anchor member with the results of the "recursive member", which is `SELECT value + 1 FROM numbers WHERE value < 10`.
+* The recursion continues until `value < 10` returns false, at which point the CTE stops executing.
+
+## Using CTEs to modify information in a table
+
+```{important}
+Without a unique identifier for each row in your table, performing UPDATE or DELETE operations using a CTE would be risky because they could affect more rows than you intend. 
+```
+
+It's generally not recommended to use UPDATE or DELETE without a unique identifier or precise condition to pinpoint exactly which rows you want to affect. We're going to load data from the [Joining Data in SQL tutorial](https://ploomber-sql.readthedocs.io/en/latest/intro-to-sql/joining-data-in-sql.html#load-the-data) as this data has tables with unique ids.
```
Comment (TegveerG) [inline course-material/advanced-querying-techniques/ctes.md:202]: ids -> ID's

File: course-material/advanced-querying-techniques/ctes.md [hunk 2]
```diff
@@ -6,9 +6,417 @@ jupytext:
...
+SELECT * 
+FROM Region_Inhabitants;
+```
+
+This CTE, `Region_Inhabitants`, groups the dataset by region and calculates the total number of inhabitants for each region using the `SUM()` function. The final SELECT statement retrieves all records from the CTE.
+
+</details>
+<!-- #endregion -->
+
+## Exercise (hard)
```
Comment (TegveerG) [inline course-material/advanced-querying-techniques/ctes.md:363]: ## Exercise 3 (hard)

**Review by TegveerG — APPROVED** [review]
Great work! Added minor suggestions

**Comment by lfunderburk** [comment]
closing due to refactor, opening new pr from latest main banch

**Pull request closed without merging**